lnwallet/channel.go

```diff
@@ -1328,19 +1342,15 @@ func (lc *LightningChannel) ReceiveNewCommitment(rawSig []byte,
 	return nil
 }
 
-// PendingUpdates returns a boolean value reflecting if there are any pending
-// updates which need to be committed. The state machine has pending updates if
-// the local log index on the local and remote chain tip aren't identical. This
-// indicates that either we have pending updates they need to commit, or vice
-// versa.
-func (lc *LightningChannel) PendingUpdates() bool {
+// NeedUpdate returns a boolean value reflecting if there are any pending
+// updates which need to be committed. The commitment transaction should
+// be updated if we have htlcs which are not committed in remote chain.
+func (lc *LightningChannel) NeedUpdate() bool {
 	lc.RLock()
 	defer lc.RUnlock()
 
-	fullySynced := (lc.localCommitChain.tip().ourMessageIndex ==
-		lc.remoteCommitChain.tip().ourMessageIndex)
-
-	return !fullySynced
+	return	lc.remoteCommitChain.tip().ourMessageIndex != lc.ourLogCounter ||
```

Comment:
First of all I know that we talked about this method a lot of time, but I wasn't able to understand why this change is wrong, because actually is not the change it is just another method which answers on another questi. For me instead it seems quite prudent for several reasons: 

As stated in the commit message I change the meaning of this method completely because it was used only in one place, so again it more rewrite than a change:
> Now 'NeedUpdate' function return 'true' if we need to send the commitment
> signature message to remote peer. In previous implementation function
> returns 'true' if nodes are not synced with each other, which is another
> meaning.

In current implementation I use this method inside the `updateCommitTx` method and as far as the logic of `updateCommitTx` implies that we:
- generate new commitment transaction
- include all htlc updates in it
- add it to the remote chain
- send it to the remote side

The goal of `NeedUpdate` in this case is to answer on question before trying to update the commit tx - Do we have any updates that are not included in last commitment transaction that were sent by us or to us?

So, again sorry for wasting so much time on discussing it.